### PR TITLE
Add Multi TXs to TX History

### PR DESCRIPTION
**Pull request by FrederikBolding**
Steps to test:
- Go through a Multi TX flow (membership, rep migration etc)
- Refrain from refreshing until the test is done (to make sure you don't fetch the tx's from etherscan)
- Go to the dashboard
- Check whether the transactions you just made are in the tx history

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch6245]

File: src/utils/useTxMulti/useTxMulti.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+import { waitFor } from 'test-utils';
+
+import { fAccount, fAccounts, fAssets, fNetwork, fNetworks } from '@fixtures';
+import { ITxObject, ITxToAddress, ITxValue, ITxData, ITxHash, ITxType, ITxStatus } from '@types';
+import { AccountContext, DataContext, IDataContext, StoreContext } from '@services';
+
+import { useTxMulti } from './useTxMulti';
+
+const createTxRaw = (idx: number): Partial<ITxObject> => ({
+  to: ('address' + idx) as ITxToAddress,
```
Comment (vilsbole) [inline src/utils/useTxMulti/useTxMulti.spec.tsx:13]: Why `ITxToAddress` instead of `TAddress`?

File: src/utils/useTxMulti/useTxMulti.ts [hunk 2]
```diff
@@ -30,6 +34,28 @@ export type TUseTxMulti = () => {
...
+
+  useEffect(() => {
+    if (
+      account &&
+      currentTx &&
+      currentTx.txResponse &&
+      currentTx.txHash &&
+      currentTx.status === ITxStatus.BROADCASTED
+    ) {
+      const network = getNetworkByChainId(currentTx.txRaw.chainId)!;
```
Comment (vilsbole) [inline src/utils/useTxMulti/useTxMulti.ts:53]: Isn't `network` already available in `state`?

File: src/utils/useTxMulti/useTxMulti.ts [hunk 3]
```diff
@@ -30,6 +34,28 @@ export type TUseTxMulti = () => {
...
+      const txConfig = makeTxConfigFromTxResponse(currentTx.txResponse, assets, network, accounts);
+      const pendingTxReceipt = makePendingTxReceipt(currentTx.txHash)(ITxType.UNKNOWN, txConfig);
+      addNewTxToAccount(account, pendingTxReceipt);
+    }
+  }, [currentTx]);
```
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/utils/useTxMulti/useTxMulti.ts:58]: It seems that you rely on `assets, accounts`. In which case they should be added to the dependencies.

File: src/utils/useTxMulti/useTxMulti.ts [hunk 4]
```diff
@@ -30,6 +34,28 @@ export type TUseTxMulti = () => {
 export const useTxMulti: TUseTxMulti = () => {
   const [state, dispatch] = useReducer(TxMultiReducer, initialState, identity, 'TxMulti');
   const getState = () => state;
+  const { accounts } = useContext(StoreContext);
+  const { addNewTxToAccount } = useContext(AccountContext);
```
Comment (vilsbole) [inline src/utils/useTxMulti/useTxMulti.ts:38]: Naming: `addNew` is an oxymore. Prefer `addTxToAccount`.

File: src/utils/useTxMulti/useTxMulti.ts [hunk 5]
```diff
@@ -30,6 +34,28 @@ export type TUseTxMulti = () => {
...
+      account &&
+      currentTx &&
+      currentTx.txResponse &&
+      currentTx.txHash &&
+      currentTx.status === ITxStatus.BROADCASTED
+    ) {
+      const network = getNetworkByChainId(currentTx.txRaw.chainId)!;
+      const txConfig = makeTxConfigFromTxResponse(currentTx.txResponse, assets, network, accounts);
+      const pendingTxReceipt = makePendingTxReceipt(currentTx.txHash)(ITxType.UNKNOWN, txConfig);
+      addNewTxToAccount(account, pendingTxReceipt);
```
Comment (vilsbole) [inline src/utils/useTxMulti/useTxMulti.ts:56]: It's really odd that we need so many manipulations between formats.
Can me converge the them?

**Review by vilsbole — COMMENTED** [review]
I suspect this adds multiple txReceipts.

File: src/utils/useTxMulti/useTxMulti.spec.tsx (hunk 1 above)
Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.spec.tsx:13]: That's the type on `ITxObject`

File: src/utils/useTxMulti/useTxMulti.ts (hunk 2 above)
Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.ts:53]: I hadn't spotted that, but yeah!

File: src/utils/useTxMulti/useTxMulti.ts (hunk 4 above)
Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.ts:36]: Let's rename when we refactor `AccountProvider`

Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.ts:36]: Is part of 3508

File: src/utils/useTxMulti/useTxMulti.ts (hunk 5 above)
Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.ts:53]: I think this is a general thing that we have to look into. But very much out of the scope of this PR.

File: src/utils/useTxMulti/useTxMulti.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.ts:55]: Perhaps it doesn't need dependencies.

[the remark this answers is not in the document]
Comment (FrederikBolding) [reply src/utils/useTxMulti/useTxMulti.ts:55]: Actually it changes the amount of times run with/without the current dependency.

File: src/utils/useTxMulti/useTxMulti.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
+import React from 'react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import isEmpty from 'ramda/src/isEmpty';
```
Comment (vilsbole) [inline src/utils/useTxMulti/useTxMulti.spec.tsx:3]: from `@vendor`

**Review by vilsbole — APPROVED** [review]
Code lgtm;
Merge waiting for additional testing.

**Pull request merged**